### PR TITLE
Sync v2-8-stable with v2-8-test to release 2.8.0

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.8.0b1`!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,7 +21,198 @@
...
+- Improve importing the modules in Airflow macros, metrics, notifications and secrets packages (#33811)
+- Improve importing the modules in Airflow www package (#33810)
+- Improve importing the modules in Airflow cli package (#33806)
+- Improve importing the modules in Airflow auth, api_internal and callbacks packages (#33807)
+- Improve importing the modules in Airflow jobs package (#33805)
+- Improve importing the module in Airflow decorators package (#33804)
+- Improve importing the module in Airflow utils package (#33803)
+- Improve importing the module in Airflow sensors package (#33801)
+- Improve importing the module in Airflow models package (#33799)
+- Improve importing the module in Airflow operators package (#33800)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]: I will still work on this changelog...

File: README.md [hunk 2]
```diff
@@ -90,13 +90,13 @@ Airflow is not a streaming solution, but it is often used to process real-time d
 
 Apache Airflow is tested with:
 
-|             | Main version (dev)     | Stable version (2.7.3)       |
+|             | Main version (dev)     | Stable version (2.8.0)       |
 |-------------|------------------------|------------------------------|
 | Python      | 3.8, 3.9, 3.10, 3.11   | 3.8, 3.9, 3.10, 3.11         |
 | Platform    | AMD64/ARM64(\*)        | AMD64/ARM64(\*)              |
 | Kubernetes  | 1.25, 1.26, 1.27, 1.28 | 1.24, 1.25, 1.26, 1.27, 1.28 |
```
Comment (raphaelauv) [inline README.md]: 1.24 is not include in 2.8.0

File: README.md [hunk 3]
```diff
@@ -90,13 +90,13 @@ Airflow is not a streaming solution, but it is often used to process real-time d
...
-|             | Main version (dev)     | Stable version (2.7.3)       |
+|             | Main version (dev)     | Stable version (2.8.0)       |
 |-------------|------------------------|------------------------------|
 | Python      | 3.8, 3.9, 3.10, 3.11   | 3.8, 3.9, 3.10, 3.11         |
 | Platform    | AMD64/ARM64(\*)        | AMD64/ARM64(\*)              |
-| Kubernetes  | 1.25, 1.26, 1.27, 1.28 | 1.24, 1.25, 1.26, 1.27, 1.28 |
-| PostgreSQL  | 11, 12, 13, 14, 15, 16 | 11, 12, 13, 14, 15           |
-| MySQL       | 8.0, Innovation        | 5.7, 8.0                     |
+| Kubernetes  | 1.25, 1.26, 1.27, 1.28 | 1.25, 1.26, 1.27, 1.28       |
+| PostgreSQL  | 11, 12, 13, 14, 15, 16 | 11, 12, 13, 14, 15, 16       |
```
Comment (DjVinnii) [inline README.md]: Is PostgreSQL 11 still supported in 2.8? It is EOL since the 9th of November and #35785 also removed it from the docs.

**Review by hussein-awala — APPROVED** [review]
Looks good

**Review by potiuk — APPROVED** [review]

**Pull request merged**